### PR TITLE
feat: post a pr to bcr on successful release

### PR DESCRIPTION
**Pull request by kormide**
The plan is to turn this into an actions plugin and host on `bazel-contrib`, but want to test run it in bazel-lib first.

### CODE REVIEW

File: .github/workflows/create-bcr-entry.mjs [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+
+  writeFileSync(destPath, newSource, {
+    encoding: "utf-8",
+  });
+}
+
+function normalizeVersion(version) {
+  if (version.startsWith("v")) {
+    return version.substring(1);
+  }
```
Comment (kormide) [inline .github/workflows/create-bcr-entry.mjs:139]: TODO: Need to differentiate between github tag and bcr version and not make assumptions.

File: .bcr/metadata.template.json [hunk 2]
```diff
@@ -0,0 +1,12 @@
+{
```
Comment (alexeagle) [inline .bcr/metadata.template.json:1]: I think I'd rather keep these files next to the bcr.yml in workflows, like where the workspace_snippet.sh is

**Review by alexeagle — APPROVED** [review]
nice approach to avoid using any data handed to us by GH actions, so we could even mirror releases of repos we don't contribute to.

**Comment by kormide** [comment]
To get this working one of needs to set up a github Personal Access Token (PAT) with repo scope, then add a secret actions variable for `bazel-lib` called `AUTO_BCR_TOKEN`.

File: .github/workflows/bcr/source.template.json [hunk 3]
```diff
@@ -0,0 +1,5 @@
+{
+  "integrity": "**leave this alone**",
+  "strip_prefix": "{REPO}-{VERSION}",
+  "url": "https://github.com/{OWNER}/{REPO}/archive/{TAG}.tar.gz"
```
Comment (kormide) [inline .github/workflows/bcr/source.template.json:4]: Some projects use releases, others use tag refs to refer to the archive, so this file supports multiple variable substitutions to support that. Will be documented when this becomes a proper plugin.

Comment (kormide) [reply .github/workflows/bcr/source.template.json:4]: Whoever's PAT is used will be the author of the PR. I'm not able to set this up as I don't have the permissions in our org.

File: .github/workflows/bcr.yml [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+name: BCR
+
+on:
+  release:
+    types: [published]
+
+env:
+  # Bazel central registry to post a pull request to.
+  BCR: aspect-build/bazel-central-registry
```
Comment (kormide) [inline .github/workflows/bcr.yml:25]: This is set up to post to our fork currently.

File: .github/workflows/bcr.yml [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+#   2. Set the workflow env variables below (see instructions for each variable).
+#   3. Update the following files in .github/workflows/bcr to be relevant for your
+#      project:
+#         - presubmit.yml
+#         - metadata.template.json
+#         - source.template.json
+#      See https://docs.bazel.build/versions/main/bzlmod.html#bazel-central-registry
+#      for more information.
+#   4. A PR will be posted from your fork to bazelbuild/bazel-central-registry
+#      with an updated module entry after a successful release.
```
Comment (kormide) [inline .github/workflows/bcr.yml:15]: Some of these docs aren't correct for our use case right now---publishing to our own for `aspect-build/bazel-central-registry`.

File: .github/workflows/bcr.yml [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+          ref: ${{ env.GITHUB_REF }}
+          path: project
+      - name: Checkout bcr
+        uses: actions/checkout@v3
+        with:
+          repository: ${{ env.BCR }}
+          path: bcr
+      - name: Create bcr entry
+        run: node .github/workflows/create-bcr-entry.mjs project bcr $GITHUB_REPOSITORY ${{ steps.get_tag.outputs.TAG }}
+      - name: Create Pull Request
```
Comment (alexeagle) [inline .github/workflows/bcr.yml:62]:
```
      - uses: tibdex/github-app-token@v1
        id: generate-token
        with:
          app_id: ${{ secrets.BCR_APP_ID }}
          private_key: ${{ secrets.BCR_APP_PRIVATE_KEY }}
```

File: .github/workflows/bcr.yml [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+        id: post_pr
+        uses: peter-evans/create-pull-request@v3
+        with:
+          token: ${{ env.TOKEN }}
```
Comment (alexeagle) [inline .github/workflows/bcr.yml:66]: token: ${{ steps.generate-token.outputs.token }}

File: .github/workflows/bcr.yml [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+      - name: Checkout bcr
+        uses: actions/checkout@v3
+        with:
+          repository: ${{ env.BCR }}
+          path: bcr
+      - name: Create bcr entry
+        run: node .github/workflows/create-bcr-entry.mjs project bcr $GITHUB_REPOSITORY ${{ steps.get_tag.outputs.TAG }}
+      - name: Create Pull Request
+        id: post_pr
+        uses: peter-evans/create-pull-request@v3
```
Comment (alexeagle) [inline .github/workflows/bcr.yml:64]: there is a v4 of this now

**Pull request merged**